### PR TITLE
Deprecated API in openvino/core/validation_util.hpp

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Deprecated API in openvino/core/validation_util.hpp
 - *...*

### Tickets:
 - CVS-106449

### CODE REVIEW

File: src/tests/functional/plugin/conformance/subgraphs_dumper/src/utils/node.cpp [hunk 1]
```diff
@@ -116,19 +116,17 @@ std::shared_ptr<ov::Node> clone_node(std::shared_ptr<ov::Node> node,
     for (size_t i = 0; i < node->get_input_size(); ++i) {
         std::string input_name = node_name + "_" + std::to_string(i);
         // todo: replace deprecated code && remove this w/a for constant size
-        OPENVINO_SUPPRESS_DEPRECATED_START
-        const auto constant_input = ov::get_constant_from_source(node->input(i).get_source_output());
```
Comment (zhaixuejun1993) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/src/utils/node.cpp:120]: @ilya-lavrenov  @vurusovs  the API ov::get_constant_from_source() will be deprecated, ov::util::get_constant_from_source() from dev_api can not be used here, any suggestions?

**Comment by zhaixuejun1993** [comment]
Blocked by PR: https://github.com/openvinotoolkit/openvino_contrib/pull/855

File: src/tests/functional/plugin/conformance/subgraphs_dumper/src/utils/node.cpp (hunk 1 above)
Comment (riverlijunjie) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/src/utils/node.cpp:120]: Why `ov::util::get_constant_from_source()` cannot be used?

File: samples/cpp/common/utils/include/samples/classification_results.h [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #include <utility>
 #include <vector>
 
+#include "openvino/op/constant.hpp"
```
Comment (ilya-lavrenov) [inline samples/cpp/common/utils/include/samples/classification_results.h:18]: I don't see that `Constant` is used in this file

File: samples/cpp/common/utils/include/samples/classification_results.h [hunk 3]
```diff
@@ -15,6 +15,7 @@
 #include <utility>
 #include <vector>
 
+#include "openvino/core/type/element_type_traits.hpp"
```
Comment (ilya-lavrenov) [inline samples/cpp/common/utils/include/samples/classification_results.h:18]: is it not accessible from `openvino/openvino.hpp` ?
maybe we can add it there then?

Comment (zhaixuejun1993) [reply samples/cpp/common/utils/include/samples/classification_results.h]: Great

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by zhaixuejun1993** [comment]
Blocked by VPUX Plugin PR: https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/pull/8675/

**Review by rkazants — APPROVED** [review]
FEs changes look good.

**Comment by zhaixuejun1993** [comment]
Blocked by: https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/pull/8730

**Review by vurusovs — APPROVED** [review]

**Pull request merged**